File: twoSum.js
/**
 * @param (number[]) nums
 * @param (number) target
 * @return (number[])
 */

const twoSum = function(nums,target) {
    let storedValues = {};

    for (let [i,num] of nums.entries()) {
        if (storedValues[num] !== undefined) {
            return [storedValues[num], i];
        }
        storedValues[target - num] = i;
    }
};

File: twosum.go
package main

func twoSum(nums []int, target int) []int {
	stored := map[int]int{}

	for i, num := range nums {
		number := num
		if index, found := stored[number]; found {
			return []int{index, i}
		}
		stored[target-num] = i
	}
	return nil
}
